src/core/server/saved_objects/mappings/lib/get_root_properties_objects.ts

```diff
@@ -30,18 +31,20 @@ import { getRootProperties } from './get_root_properties';
  *  This data can be found at `{indexName}.mappings.{typeName}.properties`
  *  in the es indices.get() response where the properties are objects.
  *
- *  @param  {EsMappingsDsl} mappings
- *  @return {EsPropertyMappings}
+ *  @param  {EsMappings} mappings
+ *  @return {EsMappings}
  */
-export function getRootPropertiesObjects(mappings) {
+export function getRootPropertiesObjects(mappings: EsMappings): EsMappings {
   const rootProperties = getRootProperties(mappings);
-  return Object.entries(rootProperties).reduce((acc, [key, value]) => {
+  return Object.entries(rootProperties).reduce(
+    (acc: EsMappings, [key, value]) => {
+      // if value has properties or a type of object, we assume it is an object datatype
+      if (value.properties || value.type === 'object') {
+        acc[key] = value;
+      }
 
-    // we consider the existence of the properties or type of object to designate that this is an object datatype
-    if (value.properties || value.type === 'object') {
-      acc[key] = value;
-    }
-
-    return acc;
-  }, {});
+      return acc;
+    },
+    {} as any
+  ) as EsMappings;
```

Comment:
No logic changes here. I think the return type of EsMappings might be wrong. Maybe the type is actually { [key: string]: EsMapping } because it's a collection of the root es type's properties. Each property is an EsMapping.

Reply:
Yeah, I feel like it's a little confusing because the total mapping definition object (`EsMappings`) is structured the same was as the `properties` value of an object mapping, so TypeScript isn't going to treat them differently but maybe we want to have two types so that the type named better describe what the value actually is.